staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -464,14 +464,25 @@ func testSetupWithEtcdServer(t *testing.T, opts ...setupOption) (context.Context
 		t.Fatalf("Failed to inject list errors: %v", err)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
+		// The tests assume that Get/GetList/Watch calls shouldn't fail.
+		// However, 429 error can now be returned if watchcache is under initialization.
+		// To avoid rewriting all tests, we wait for watcache to initialize.
+		if err := cacher.ready.wait(ctx); err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
These tests are currently run with the feature gate disabled?  Should we run them with the gate both disabled and enabled?

Reply:
I'm not sure there is a big value in it. The tests in this package are relatively close to the time limit already and we can't afford duplicating them without splitting package itself. 

What we can potentially do is add a simple test that ensures that we won't get 429 error from Get, List, Watch on startup when FG is disabled [and we will be able to easily remove this test on graduation to GA].
Would that work for you?

Reply:
That sounds good to me

Reply:
Added TestTooManyRequestsNotReturned